src/ValueParsers/MatrixParser.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace ValueParsers;
+
+use DataValues\DecimalValue;
+use DataValues\IllegalValueException;
+use DataValues\MatrixValue;
+use InvalidArgumentException;
+
+/**
+ * ValueParser that parses the string representation of a matrix.
+ *
+ * @licence GNU GPL v2+
+ * @author Andrius Merkys < andrius.merkys@gmail.com >
+ */
+class MatrixParser extends StringValueParser {
+
+	const FORMAT_NAME = 'matrix';
+
+	/**
+	 * @see StringValueParser::stringParse
+	 *
+	 * @param string $value
+	 *
+	 * @return MatrixValue
+	 * @throws ParseException
+	 */
+	protected function stringParse( $value ) {
+		if ( !is_string( $value ) ) {
+			throw new InvalidArgumentException( '$value must be a string' );
+		}
+
+		$decoded = json_decode( $value );
```

Comment:
JSON seems cumbersome for human input/output. I recommend CSV:

  1, 2
  3, 4

There is plenty of code for CSV out there, so you don't have to roll your own.